generate/templates/all.template

```diff
@@ -13,6 +14,22 @@ func AllResources() map[string]Resource {
 }
 
 {{range $resource := .Resources}}
+{{if $resource.Global}}
+// GetServerlessGlobal{{$resource.StructName}}retrieves the template's Globals.{{$resource.StructName}} items from an AWS SAM template.
+// Returns an error if not found.
+func (t *Template) GetServerlessGlobal{{$resource.StructName}} () (*global.{{$resource.StructName}}, error) {
+	globals := *t.Globals
```

Comment:
This fails:
 - `t.Globals undefined (type *Template has no field or method Globals)`
 - It should also probably be `t.Globals`, not `*t.Globals` 

I think for this PR to be merged, we need to be able to parse a CloudFormation template with globals into Go objects. I know that's not your primary motivation for this PR, but it's what people will expect.

Given the below template:

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: Testing Globals

Globals:
  Function:
    Runtime: nodejs12.x
    Timeout: 180
    Handler: index.handler
    Environment:
      Variables:
        TABLE_NAME: data-table

Resources:
  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      Environment:
        Variables:
          MESSAGE: "Hello From SAM"
```

I would expect to be able to parse it with `goformation.Open(filename)`, and inspect the global elements like so:

```go
package main

import (
	"log"

	"github.com/awslabs/goformation/v4"
	"github.com/davecgh/go-spew/spew"
)

func main() {

	template, err := goformation.Open("test/globals.yml")
	if err != nil {
		log.Fatalf("There was an error processing the template: %s\n", err)
	}

	gf, err := template.GetServerlessGlobalFunction()
	if err != nil {
		log.Fatalf("No global function set")
	}

	spew.Dump(gf)

}
```

I've been trying to this how best to implement this. If we add a `Globals` field to the `Template` struct, it needs to be able to support any of (and only) the Global resource types, but Go has no sum types. So we'd need to create an interface.

In `cloudformation/template.go`:
```go
type Template struct {
	AWSTemplateFormatVersion string                 `json:"AWSTemplateFormatVersion,omitempty"`
	Transform                *Transform             `json:"Transform,omitempty"`
	Description              string                 `json:"Description,omitempty"`
	Metadata                 map[string]interface{} `json:"Metadata,omitempty"`
	Parameters               Parameters             `json:"Parameters,omitempty"`
	Mappings                 map[string]interface{} `json:"Mappings,omitempty"`
	Conditions               map[string]interface{} `json:"Conditions,omitempty"`
	Resources                Resources              `json:"Resources,omitempty"`
	Outputs                  Outputs                `json:"Outputs,omitempty"`
	Globals                  Globals                `json:"Globals,omitempty"`
}

type Resource interface {
	AWSCloudFormationType() string
}

type Global interface {
	AWSCloudFormationIsGlobal() bool
}

type Resources map[string]Resource
type Globals map[string]Global

```

In `generate/templates/global.template` you'd just need to add the following method at the bottom of all generated Global types, to satisfy the interface:

```go
// AWSCloudFormationIsGlobal is used to compile an interface (cloudformation.Global) for all global resources
func (r *{{.StructName}}) AWSCloudFormationIsGlobal() bool {
    return true
}
```

note: the method above, like `AWSCloudFormationType` is prefixed with AWSCloudFormation to prevent conflicts if a CFN property ever has a matching name.

When it comes to unmarshalling the JSON/YAML into the `cloudformation/global/*` structs, it will a small bit of logic to detect which struct to unmarshal it into. This should be easy enough though, as the Logical ID is a fixed list (Function|Api|HttpApi|SimpleTable).

Reply:
Committed my changes to template.go. Check to see if that fits the bill? Definitely open to moving stuff around after that.